RadixWallet/Clients/DeviceFactorSourceClient/DeviceFactorSourceClient+Live.swift

```diff
@@ -71,68 +73,90 @@ extension DeviceFactorSourceClient: DependencyKey {
-				case let .unsecured(value):
-					mnemonicMissingFactorSources.contains(value.transactionSigning.factorSourceId)
-				}
-			}
+				let mnemomincPresentFactorSources = factorSources
+					.filter(\.present)
+					.map(\.id)
 
-			func mnemonicMissing(_ persona: Persona) -> Bool {
-				switch persona.securityState {
-				case let .unsecured(value):
-					mnemonicMissingFactorSources.contains(value.transactionSigning.factorSourceId)
-				}
-			}
+				let unrecoverableFactorSources = mnemomincPresentFactorSources
+					.filter { !backedUpFactorSources.contains($0) }
 
-			func unrecoverable(_ account: Account) -> Bool {
-				switch account.securityState {
-				case let .unsecured(value):
-					unrecoverableFactorSources.contains(value.transactionSigning.factorSourceId)
+				let network = try profile.network(id: profile.networkID)
+				let accounts = network.getAccounts()
+				let hiddenAccounts = network.getHiddenAccounts()
+				let personas = network.getPersonas()
+				let hiddenPersonas = network.getHiddenPersonas()
+
+				func mnemonicMissing(_ account: Account) -> Bool {
+					switch account.securityState {
+					case let .unsecured(value):
+						mnemonicMissingFactorSources.contains(value.transactionSigning.factorSourceId)
+					}
 				}
-			}
 
-			func unrecoverable(_ persona: Persona) -> Bool {
-				switch persona.securityState {
-				case let .unsecured(value):
-					unrecoverableFactorSources.contains(value.transactionSigning.factorSourceId)
+				func mnemonicMissing(_ persona: Persona) -> Bool {
+					switch persona.securityState {
+					case let .unsecured(value):
+						mnemonicMissingFactorSources.contains(value.transactionSigning.factorSourceId)
+					}
 				}
-			}
 
-			let mnemonicMissing = ProblematicAddresses(
-				accounts: accounts.filter(mnemonicMissing(_:)).map(\.address),
-				hiddenAccounts: hiddenAccounts.filter(mnemonicMissing(_:)).map(\.address),
-				personas: personas.filter(mnemonicMissing(_:)).map(\.address),
-				hiddenPersonas: hiddenPersonas.filter(mnemonicMissing(_:)).map(\.address)
-			)
+				func unrecoverable(_ account: Account) -> Bool {
+					switch account.securityState {
+					case let .unsecured(value):
+						unrecoverableFactorSources.contains(value.transactionSigning.factorSourceId)
+					}
+				}
 
-			let unrecoverable = ProblematicAddresses(
-				accounts: accounts.filter(unrecoverable(_:)).map(\.address),
-				hiddenAccounts: hiddenAccounts.filter(unrecoverable(_:)).map(\.address),
-				personas: personas.filter(unrecoverable(_:)).map(\.address),
-				hiddenPersonas: hiddenPersonas.filter(unrecoverable(_:)).map(\.address)
-			)
+				func unrecoverable(_ persona: Persona) -> Bool {
+					switch persona.securityState {
+					case let .unsecured(value):
+						unrecoverableFactorSources.contains(value.transactionSigning.factorSourceId)
+					}
+				}
 
-			return (mnemonicMissing: mnemonicMissing, unrecoverable: unrecoverable)
+				let mnemonicMissing = ProblematicAddresses(
```

Comment:
Hmm an address cannot be "problematic". Better `AddressesOfEntitiesInBadState` IMO.

Maybe `entitiesWithoutControl` is better than `mnemonicMissing`? Also suitable post-MFA.